### PR TITLE
#120: API core view realloc transition

### PR DESCRIPTION
**Pull request by marcinwrobel1986**
### THIS PR:
* API/core/view/realloc transition from .md to .rst

before (.md) | after (.rst)
:----------------: | :-------------:
image | image
image | ---
image | image
image | image
image | image

### CODE REVIEW

**Comment by antoinemeyer5** [comment]
> please remove synopsis , use sphinx syntax where possible and make sure you use the correct indentation for each function

- [x] Remove synopsis
- [x] Fix indentation before Restrictions lines
- [x] Make the attributes of the third function consistent

Not possible to use sphinx syntax because of ``...``

File: docs/source/API/core/view/realloc.rst [hunk 1]
```diff
@@ -0,0 +1,156 @@
+``realloc``
+===========
+
+Header File: ``<Kokkos_Core.hpp>``
+
+.. role:: cppkokkos(code)
+    :language: cppkokkos
+
```
Comment (ajpowelsnl) [inline docs/source/API/core/view/realloc.rst:8]: Please move towards a standardized structure for API documentation entries:  function name, Header File, (human language) Description, Usage and Example.

File: docs/source/API/core/view/realloc.rst [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  - ``View<T, P...>::array_layout`` is either ``LayoutLeft`` or ``LayoutRight``.
+
+  - ``arg_prop`` must not include a pointer to memory, a label, or a memory space.
+
+.. code-block:: cpp
+
+   template <class T, class... P>
+   void realloc(Kokkos::View<T, P...>& v,
+		const typename Kokkos::View<T, P...>::array_layout& layout);
+
```
Comment (ajpowelsnl) [inline docs/source/API/core/view/realloc.rst:103]: There's a lot of redundancy that can be eliminated.  Why is "* Resizes ``v`` to have the new dimensions without preserving its contents." repeated?  This sentence can be moved to the Description and said once.

**Review by ajpowelsnl — APPROVED** [review]

File: docs/source/API/core/view/realloc.rst (hunk 2 above)
Comment (fnrizzi) [reply docs/source/API/core/view/realloc.rst:103]: we are documenting each method. IMO each method should be able to stand on its own.

**Review by fnrizzi — APPROVED** [review]

**Pull request merged**